### ISSUE TITLE
Taxonomy concept enhancement - set default concept - phase 2

### ISSUE THREAD

**Issue by troughc** [culturecreates/footlight-app#1457]
See https://www.figma.com/design/DBnSSMT9Etfm6guDKKSp8O/Culture-Creates---Footlight-Calendar?node-id=11193-74280&t=zJtuT9Pep9N1Ofuj-1
This ticket adds the ability to set a concept as the default concept. This will allow the admin to always have this default selected in the associated edit form (in the associated workspace)
To do: 
- [x] Add the Set as default option (and Remove as default option) to the context menu. When you use this option, the concept becomes the default (or is removed from being the default)
- [x] Add star
- [x] make the default concept the pre-selected concept in the form (prefill the field).

**Comment by sahalali** [comment]
Backend changes: 4H
1. Add support to persist 'isDefault' property for concepts. - 2H
2. If default concepts need to be added to events via the aggregator, that will also require a change. - 2H

Question:
1. Can there be multiple default concepts under a taxonomy? 

2. Should these default concepts be applied to events coming through the aggregator? Could you consider an option in the aggregator that uses a mapping file to set a default concept? Which should take priority if a default concept is set in both the CMS and the aggregator mapping?

**Comment by SyamBabu-M** [comment]
FE estimate:

1. Check for "isDefault" boolean and its integration into the concept table. - 6h
2. Make the default concept the pre-selected concept in all the forms. - 4h

**Comment by sahalali** [comment]
Question: What is the rule incase if the taxonomy is admin only? The default concept should only be pre filled if the user is an Admin. Otherwise the concept should not be pre filled. 

@troughc

**Comment by troughc** [comment]
Happy birthday @sahalali !!!! :)


> Question: What is the rule incase if the taxonomy is admin only? The default concept should only be pre filled if the user is an Admin. Otherwise the concept should not be pre filled.

Another way of thinking about this is that if the the taxonomy is for Admin only, then ONLY the Admin will see it in the form(s). In other words, you can prefill the default concept and ONLY the admin would see it (it is hidden for all other users).

**Comment by troughc** [comment]
@sahalali I am adding this issue to the To do column. @SyamBabu-M can start work once you have started the BE work. Thanks!

**Comment by troughc** [comment]
Discussion points
- default concept is added to a field at CREATION in CMS.  This means that it is added when you are creating something from scratch (the field is pre-filled) AND when you are importing an entity from Artsdata/external source (the default is added to the related field). 
- If user removes default concept and then saves the entity, the default concept is gone (do not re-add it to the field)
- If at CREATION< a field has another value (a different concept) you still add the default concept. For example if Audience type has general audiece as a default concept, you always add it no matter what else is in the field. Once you save, it is as the user set it.
-

**Comment by SyamBabu-M** [comment]
@troughc I encountered an existing bug when importing entities from external sources, which is a blocker for this issue. Therefore, I am fixing it as part of this issue itself.

Steps to reproduce - 

1. Go to Tout culture in staging.
2. Go to [Search people page](https://staging.cms.footlight.io/dashboard/6308ef4a7f771f00431d939a/people/search).
3. Search for 'c'.
4. "Bryan Cheng" will be in the results. Click on it to import.

Result - 

<img width="1440" height="900" alt="Image" src="https://github.com/user-attachments/assets/f3d42333-e3b1-4492-9713-ac9f4eafbeca" />


cc @AbhishekPAnil

**Comment by troughc** [comment]
Tested child taxonomy is working well. PASSED

**Comment by troughc** [comment]
Tested that this works with creating an event and importing an event. PASSED

**Comment by troughc** [comment]
This is no longer deployed in production. FYI I announced this one to the users.

**Comment by sahalali** [comment]
Everything's working fine in production; I checked it out with @SyamBabu-M. It might just be a caching thing.

**Comment by troughc** [comment]
agreed - refresh fixed this (logging out and in did not - which was my original approach)
